### ISSUE TITLE
Test hangs indefinitely when encountering # in unresolvable url

### ISSUE THREAD

**Issue by Seeker9889** [DevExpress/testcafe#2371]
### Are you requesting a feature or reporting a bug?

Bug

### What is the current behavior?

A fixture tries to navigate to a url which includes a fragment identifier (in my case, for a single-page application). Because the url is misspelled or (in the case of localhost) a local server is not running, test execution does not proceed past
```
Running tests in:
 - Chrome 66.0.3359 / Windows 10.0.0

 NameOfFixture
```

This text is displayed, and the test hangs indefinitely, regardless of what tests exist. No other fixture is switched to, and the -q argument for quarantine mode does not cause multiple test attempts. 

### What is the expected behavior?

An error is expected to be reported such as
```
 Error in Role initializer -
      Failed to find a DNS-record for the resource at "https://www.splooo.com/#unimportant".
```
and testcafe will proceed to another fixture.

Note that if the # fragment identifier is removed, the error appears in this way, and another fixture is proceeded to as expected

### How would you reproduce the current behavior (if this is a bug)?

Attempt to run the following test code in an node.js project with `npm test`

#### Provide the test code and the tested page URL (if applicable)

Tested page URL: URL doesn't matter, as long as it cannot actually be resolved by DNS. https://localhost/#whatever produces the error as well.

Test code

```
fixture `Demonstrate problem`
	.page `https://www.splooo.com/#unimportant`

test('Not run', async t => {
    await t
		.click('button[id="unused"]');
});
```

### Specify your

* operating system: Windows 10
* testcafe version: 0.19.2
* node.js version: 5.3.0

**Comment by miherlosev** [comment]
Hi @Seeker9889 

Thanks for your bug report. I've reproduced the issue. But I need an additional time to investigate the reason of the problem.

**Comment by damorton** [comment]
Getting the same issue only recently

```
Running tests in:
 - Chrome 67.0.3396 / Mac OS X 10.13.4

 Test
 ✖ test

   1) - Error in Role initializer -
      Error on page "http://localhost:8888/#/":

      Uncaught TypeError: Cannot read property 'onBodyElementMutation' of undefined

      Browser: Chrome 67.0.3396 / Mac OS X 10.13.4

```

**Comment by AndreyBelym** [comment]
Hi @damorton, I don't think that it is the same issue, original post was about TestCafe hanging during a test without any errors. In your case, an error is reported. Could you please create a new issue and provide your test page?

**Comment by damorton** [comment]
Sure will do

**Comment by miherlosev** [comment]
I'm working on this

**Comment by AndreyBelym** [comment]
Hi @Seeker9889, the issue is fixed in the `testcafe@0.20.5-alpha.1` build. Please, try to install this build and feel free to reopen the issue if the problem isn't fixed on your machine.

**Comment by AndreyBelym** [comment]
Hi @Seeker9889, @sorin-davidoi I've just published `testcafe@0.20.5` with the fix! Please, update your TestCafe installation, and feel free to reopen the issue if the problem isn't fixed in your environment.

**Comment by valerii-cognite** [comment]
I feel the same error occurs on `0.23.3`:
```
Error in Role initializer -
      Failed to find a DNS-record for the resource at
      "http://undefined".
```
and `testcafe` just hangs without processing to the next fixture
